src/main/java/uk/ac/ic/wlgitbridge/server/Oauth2Filter.java

```diff
@@ -112,4 +112,10 @@ private void unauthorized(HttpServletResponse response) throws IOException {
         unauthorized(response, "Unauthorized");
     }
 
+    private void forbidden(HttpServletRequest request, HttpServletResponse response, String message)
```

Comment:
> Does it need to be a 403 in order to display the message?

Actually, the 403 there is pointless. If you look at the [code](https://github.com/eclipse/jgit/blob/9e5380e7fb797e83b6f6b5754249bea4cae534e7/org.eclipse.jgit.http.server/src/org/eclipse/jgit/http/server/GitSmartHttpTools.java#L175) you'll the code is ignored if we provide a String (because it always has to be a 200).

On the other hand, I checked and if we send two consecutive 401 it's not prompting for credentials twice. The second time displays a generic error:
```
m@ts:~/work/overleaf/git/229fsypwf$ git pull
Username for 'http://localhost:8000': 
Password for 'http://localhost:8000': 
fatal: Authentication failed for 'http://localhost:8000/229fsypwf/'
m@ts:~/work/overleaf/git/229fsypwf$ 
```